core/dnsserver/register.go

```diff
@@ -191,14 +191,21 @@ func groupConfigsByListenAddr(configs []*Config) (map[string][]*Config, error) {
 	groups := make(map[string][]*Config)
 
 	for _, conf := range configs {
-		addr, err := net.ResolveTCPAddr("tcp", net.JoinHostPort(conf.ListenHost, conf.Port))
-		if err != nil {
-			return nil, err
+		hosts := []string{}
+		if conf.ListenHosts == nil {
```

Comment:
when is this `nil`?

Reply:
if there is no BIND directive. I will change to have a default value of one listenHost with an empty string. will look closer to former data and will be easier to handle.